google/cloud/storage/internal/xml_builders.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/xml_builders.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/internal/make_status.h"
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+namespace {
+
+auto constexpr kMaxPartNumber = 10000U;
+
+}  // namespace
+
+StatusOr<std::shared_ptr<XmlNode>> BuildCompleteMultipartUploadXml(
+    std::map<unsigned int, std::string> const& parts) {
+  auto root = XmlNode::CreateRoot();
+  auto target_node = root->AppendTagNode("CompleteMultipartUpload");
+
+  for (auto const& p : parts) {
+    if (p.first == 0) {
+      return internal::InvalidArgumentError("part number cannot be zero",
+                                            GCP_ERROR_INFO());
+    }
+    if (p.first > kMaxPartNumber) {
+      return internal::InvalidArgumentError(
+          absl::StrCat("part number cannot be more than ", kMaxPartNumber),
+          GCP_ERROR_INFO());
+    }
+    auto part_tag = target_node->AppendTagNode("Part");
+    part_tag->AppendTagNode("PartNumber")
+        ->AppendTextNode(std::to_string(p.first));
+    part_tag->AppendTagNode("ETag")->AppendTextNode(
+        EscapeXmlContent(std::move(p.second)));
```

Comment:
`p` is const, so the move will have no effect.  Not that it matters for `EscapeXmlContent()`, whose parameter is a `const&`.  So, just remove the `std::move`, I guess.